src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/KeyStatisticService.cs

```diff
@@ -0,0 +1,192 @@
+        Guid releaseId,
+        Guid keyStatisticId,
+        KeyStatisticDataBlockUpdateRequest request)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async release =>
+                await _persistenceHelper.CheckEntityExists<KeyStatisticDataBlock>(keyStatisticId, query =>
+                    query.Where(keyStat => keyStat.ReleaseId == release.Id)))
+            .OnSuccess(async keyStat =>
+            {
+                _context.Update(keyStat);
+
+                keyStat.Trend = request.Trend;
+                keyStat.GuidanceTitle = request.GuidanceTitle;
+                keyStat.GuidanceText = request.GuidanceText;
+
+                await _context.SaveChangesAsync();
+
+                return _mapper.Map<KeyStatisticDataBlockViewModel>(keyStat);
+            });
+    }
+
+    public async Task<Either<ActionResult, Unit>> Delete(Guid releaseId, Guid keyStatisticId)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async release =>
+                await _persistenceHelper.CheckEntityExists<KeyStatistic>(keyStatisticId, query =>
+                    query
+                        .Include(keyStat => (keyStat as KeyStatisticDataBlock)!.DataBlock) // TODO Include can be removed in EES-3988
+                        .Where(keyStat => keyStat.ReleaseId == release.Id)))
+            // TODO: Can remove the below OnSuccess in EES-3988
+            // Key stats were previously associated with a KeyStatistic content section,
+            // which meant their data blocks had a ContentSectionId set. The below code
+            // ensures ContentSectionId is unset when a key stat is removed. If it isn't
+            // unset, that data block won't become available for selection again.
+            .OnSuccess(async keyStat =>
+            {
+                if (keyStat.GetType() == typeof(KeyStatisticDataBlock))
+                {
+                    var dataBlock = (keyStat as KeyStatisticDataBlock)!.DataBlock;
+                    dataBlock.ContentSection = null;
+                    dataBlock.ContentSectionId = null;
+                    _context.ContentBlocks.Update(dataBlock);
+                    await _context.SaveChangesAsync();
+                }
+                return keyStat;
+            })
+            .OnSuccessVoid(async keyStat =>
+            {
+                _context.KeyStatistics.Remove(keyStat);
+                await _context.SaveChangesAsync();
+            });
+    }
+
+    public async Task DeleteAnyAssociatedWithDataBlock(Guid releaseId, Guid dataBlockId)
+    {
+        var keyStats = await _context
+            .KeyStatisticsDataBlock
+            .Where(ks => ks.ReleaseId == releaseId && ks.DataBlockId == dataBlockId)
```

Comment:
Will a particular data block exist over different Releases?  If not, there's no need for the ReleaseId check as well as the DataBlockId check.

Reply:
Nope - removed

Reply:
There should also only be a single key stat associated with a data block, so I've changed the db query to use `Single` as well.